storage/src/robocorp/storage/__init__.py

```diff
@@ -0,0 +1,172 @@
+    return name
+
+
+def _get_asset(name: str, raise_if_missing: bool = True) -> Optional[AssetType]:
+    assets_client = _get_assets_client()
+    asset_id = _retrieve_asset_id(name)
+    exception = None
+
+    def _handle_error(resp):
+        # NOTE(cmin764): The API will return 404 if the asset can't be found on the
+        #  server, therefore don't let this raise, nor retry given this custom handler.
+        nonlocal exception
+        try:
+            assets_client.handle_error(resp)
+        except RequestsHTTPError as exc:
+            if exc.status_code == 404:
+                exception = exc
+            else:
+                raise
+
+    LOGGER.debug("Retrieving asset %r with resulted ID %r.", name, asset_id)
+    response = assets_client.get(asset_id, _handle_error=_handle_error)
+    if response.ok:
+        return response.json()
+
+    message = f"asset with name {name!r} and resulted ID {asset_id!r} couldn't be found"
+    if raise_if_missing:
+        raise AssetNotFound(message) from exception
+
+    LOGGER.warning("%s%s!", message[0].upper(), message[1:])
+    return None
+
+
+def get_asset(name: str) -> str:
+    """Get the asset's value by providing its `name`.
+
+    Args:
+        name: Name of the asset to fetch.
+
+    Returns:
+        The previously set value of this asset. (empty if it was never set)
+
+    Raises:
+        AssetNotFound: When the queried asset doesn't exist.
+    """
+    LOGGER.info("Retrieving asset %r.", name)
+    payload = _get_asset(name)["payload"]  # type: ignore
+    if payload["type"] == "empty":  # type: ignore
+        LOGGER.warning("Asset %r has no value set!", name)
+        return ""
+
+    assets_client = _get_assets_client()
+    url = payload["url"]  # type: ignore
+    return assets_client.get(url, headers={}).text
+
+
+def _create_asset(name: str):
+    LOGGER.debug("Creating new asset with name %r.", name)
+    assets_client = _get_assets_client()
+    body = {"name": name}
+    return assets_client.post("", json=body).json()
+
+
+def set_asset(name: str, value: str, wait: bool = True):
+    """Sets/Creates an asset named `name` with the provided `value`.
+
+    Args:
+        name: Name of the existing or new asset to create. (if missing)
+        value: The new value set within the asset.
+        wait: Blocks until the new value is reflected within the asset.
+
+    Raises:
+        AssetNotFound: When the queried asset doesn't exist.
+    """
+    existing_asset = _get_asset(name, raise_if_missing=False)
+    if existing_asset:
+        asset_id = existing_asset["id"]
+        LOGGER.debug("Updating already existing asset with ID %r.", asset_id)
+    else:
+        asset_id = _create_asset(name)["id"]
+        LOGGER.debug("Updating newly created asset with ID %r.", asset_id)
+
+    assets_client = _get_assets_client()
+    body = {"content_type": "text/plain"}
+    upload_data = assets_client.post(f"{asset_id}/upload", json=body).json()
```

Comment:
Question: are there limits on the size of the storage?

If we could query it maybe we could fail earlier (instead of sending all the data and then getting some exception when half-way there).

Reply:
I think there are some BIG hard limits, but not aware of them. (cc @ovatto )

We would have a 6MB limit if we'd directly send the content to the server in one go, but with the S3 pre-signed ones we're safe to assume "unlimited" for decent sizes.